### PR TITLE
navidrome: Compile server binary and frontend from source

### PR DESCRIPTION
**Pull request by anpandey**
###### Description of changes

We currently use release tarballs and binaries distributed by Navidrome. Moving to compilation from source allows e.g. compiling from a different repo or applying arbitrary patches to Navidrome.

The build process and update script is based on that of gotify-server, which similarly includes a Go backend and React frontend.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by squalus** [comment]
I have a similar PR open at https://github.com/NixOS/nixpkgs/pull/180798

File: pkgs/servers/misc/navidrome/default.nix [hunk 1]
```diff
@@ -1,48 +1,48 @@
...
 with lib;
 
-stdenv.mkDerivation rec {
+buildGoModule rec {
   pname = "navidrome";
-  version = "0.47.5";
+  version = import ./version.nix;
 
+  # Can't use -trimpath because tests require accurate working dir.
+  allowGoReference = true;
```
Comment (squalus) [inline pkgs/servers/misc/navidrome/default.nix:11]: This leaves `go` as part of the runtime closure, which adds an extra ~500MB.

Comment (anpandey) [reply pkgs/servers/misc/navidrome/default.nix:11]: Is there a way to set GOFLAGS to not include `-trimpath` so that `doCheck` can be left enabled?

**Comment by anpandey** [comment]
Probably should have checked open PRs first :). I'll leave this open for the time being.

**Pull request closed without merging**